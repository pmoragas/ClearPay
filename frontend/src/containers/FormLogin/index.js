import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { useForm } from 'react-hook-form';

import styles from './styles.module.scss';

const FormLogin = ({ onSubmitForm, className }) => {
	const { register, handleSubmit } = useForm();
	const onSubmit = (data) => {
		onSubmitForm(data);
	};

	return (
		<div className={classnames(styles.formContainer, className)}>
			<div className={styles.loginTitle}>clearpay admin</div>
			<div className={styles.formWrapper}>
				<form onSubmit={handleSubmit(onSubmit)}>
					<div className={styles.inputWrapper} >
						<label className={styles.label}>
							Email
							<input label='Email' name="email" required {...register('email')}/>
						</label>
					</div>
					<div className={styles.inputWrapper}>
						<label className={styles.label}>
							Password
							<input
								label='Password'
								name="password"
								required
								{...register('password')}
								type="password"
								className={styles.password}
							/>
						</label>

					</div>

					<div className={styles.buttonContainer}>
						<button type="submit" className={styles.button}>
							Submit
						</button>
					</div>
				</form>
			</div>
		</div>
	);
};

FormLogin.propTypes = {
	onSubmitForm: PropTypes.func.isRequired,
	className: PropTypes.string,
};

FormLogin.defaultProps = {
	className: undefined,
};

export default FormLogin;
