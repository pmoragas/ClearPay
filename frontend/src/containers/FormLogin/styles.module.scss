@import '~styles/colors';

.formContainer {
    position: absolute;
    top: 40%;
    left: 50%;
    transform: translate(-50%, -50%);
    height: 275px;
    max-width: 375px;
    width: 100%;
    font-family: sans-serif;
    display: flex;
    flex-direction: column;
    box-shadow: 0 8px 21px 0 rgba($primary, 0.16);
    border: solid 1px rgba($secondary, 0.5);
    background-color: $white;

    .loginTitle {
        padding: 15px;
        font-weight: 600;
        font-size: 20px;
        text-align: center;
        color: $tertiary;
        border-bottom: solid 1px rgba($secondary, 0.5);
        background-color: $primary;
    }

    .formWrapper {
        margin: auto;
        display: flex;
        flex-direction: column;
        align-items: center;

        form {
            padding: 15px;
            display: flex;
            flex-direction: column;
            align-items: center;
            width: 100%;

            .inputWrapper {
                padding: 5px;

                .label {
                    color: $tertiary;
                    font-size: 14px;
                }
                input {
                    width: 250px;
                    height: 20px;
                    border: solid 1px rgba($secondary, 0.5);
                    background-color: rgba($primary, 0.16);
                }
            }

            .buttonContainer {
                padding: 15px 15px 0 15px;

                button {
                    font-weight: 600;
                    font-size: 16px;
                    border: solid 1px rgba($secondary, 0.5);
                    background-color: rgba($primary, 0.16);
                    color: $tertiary;

                    &:hover {
                        background-color: rgba($secondary, 0.5);
                    }
                }
            }
        }
    }
}