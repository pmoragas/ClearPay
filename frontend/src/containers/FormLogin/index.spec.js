import React from 'react';
import { render, screen } from '@testing-library/react';
import FormLogin from './index';

describe('FormLogin', () => {
	test('Render FormLogin', () => {
		render(<FormLogin onSubmitForm={() => console.log('submitted')}/>);

		expect(screen.getByLabelText('Email')).toBeInTheDocument();
		expect(screen.getByLabelText('Password')).toBeInTheDocument();
		expect(screen.getByText('Submit')).toBeInTheDocument();
	});

	test('snapshot is ok', () => {
		const { container } = render(<FormLogin onSubmitForm={() => console.log('submitted')}/>);

		expect(container).toMatchSnapshot();
	});
});
