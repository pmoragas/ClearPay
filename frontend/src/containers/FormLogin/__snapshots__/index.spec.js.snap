// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormLogin snapshot is ok 1`] = `
<div>
  <div
    class="formContainer"
  >
    <div
      class="loginTitle"
    >
      clearpay admin
    </div>
    <div
      class="formWrapper"
    >
      <form>
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            Email
            <input
              label="Email"
              name="email"
              required=""
            />
          </label>
        </div>
        <div
          class="inputWrapper"
        >
          <label
            class="label"
          >
            Password
            <input
              class="password"
              label="Password"
              name="password"
              required=""
              type="password"
            />
          </label>
        </div>
        <div
          class="buttonContainer"
        >
          <button
            class="button"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
